specification/network/resource-manager/Microsoft.Network/stable/2022-01-01/cloudServiceVipSwap.json

```diff
@@ -0,0 +1,180 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "NetworkManagementClient",
+    "description": "The Microsoft Azure Network management API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+    "version": "2022-01-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{groupName}/providers/Microsoft.Compute/cloudServices/{resourceName}/swap/{singletonResource}": {
```

Comment:
>Compute [](http://example.com/codeflow?start=84&length=7)

why is this path being put in the Microsoft.Network spec?

Reply:
The API signature is like this - "/subscriptions/{subscriptionId}/resourceGroups/{groupName}/providers/Microsoft.Compute/cloudServices/{resourceName}/swap/{singletonResource}.
Though the API is for Microsoft.Compute but the NRP has all the implementations and in the manifest also the base URI used for this API is network.azure.com where the ARM will redirect the call to NRP. I have defined this API inside network because there is already an existing API for publicIpAddresses on cloud service which has same signature but API is defined under network folder here - https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fblob%2Fmain%2Fspecification%2Fnetwork%2Fresource-manager%2FMicrosoft.Network%2Fstable%2F2021-05-01%2FcloudServicePublicIpAddress.json%23%3A~%3Atext%3D%2522%2Fsubscriptions%2F%257BsubscriptionId%257D%2FresourceGroups%2F%257BresourceGroupName%257D%2Fproviders%2FMicrosoft.Compute%2FcloudServices%2F%257BcloudServiceName%257D%2Fpublicipaddresses%2522%253A%2520%257B&data=04%7C01%7CShraddha.Nayak%40microsoft.com%7Ce8f096debfce4969f2a608d9a96b0988%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637727099490696576%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=a8emD2Hmg7r4okelOQxCOt0IwtPgVrvujlR7ATgTCOY%3D&reserved=0
Ca we skip this too and have the API defined under Network?